test/e2e/token_inactivity_timeout_test.go

```diff
@@ -0,0 +1,284 @@
+		}
+	}
+
+	// This checks that
+	// 1. Token with timeout works immediately after they are created
+	// 2. Token with timeout works anytime before it times out
+	// 3. Token with timeout does not work after it times out
+	// 4. Token without timeout works at anytime
+	testTokenTimeoutScenarios := func(t *testing.T, tokenWithTimeout, tokenWithoutTimeout *oauthapi.OAuthAccessToken, timeout time.Duration) {
+		req, err := http.NewRequest(http.MethodGet, kubeConfig.Host+"/apis/user.openshift.io/v1/users/~", &bytes.Buffer{})
+		require.NoError(t, err)
+
+		testTokenValidity(req, http.StatusOK, tokenWithTimeout.Name, "accessing token before it timed out should work")
+		testTokenValidity(req, http.StatusOK, tokenWithoutTimeout.Name, "token with out timeout should work")
+
+		time.Sleep(120 * time.Second)
+
+		testTokenValidity(req, http.StatusOK, tokenWithTimeout.Name, "accessing token before it timed out should work")
+		testTokenValidity(req, http.StatusOK, tokenWithoutTimeout.Name, "token with out timeout should work")
+
+		time.Sleep(timeout + 10*time.Second)
+
+		testTokenValidity(req, http.StatusUnauthorized, tokenWithTimeout.Name, "accessing token after it timed out should not work")
+		testTokenValidity(req, http.StatusOK, tokenWithoutTimeout.Name, "token with out timeout should work")
+	}
+
+	configInactivityTimeout := int32(600)
+	oauthClientTimeout := int32(300)
+
+	// No OAuthClient timeout and no OAuth config timeout.
+	t.Run("without-inactivity-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, nil, testTokenTimeoutScenarios)
+	})
+
+	updateOAuthConfigInactivityTimeout(t, configClient, &metav1.Duration{Duration: 600 * time.Second})
+	test.WaitForKubeAPIServerStartProgressing(t, configClient)
+	test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	// With only OAuth config timeout.
+	t.Run("with-inactivity-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, nil, testTokenTimeoutScenarios)
+	})
+
+	// With both OAuth config timeout and OAuthClient timeout.
+	t.Run("with-client-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, &oauthClientTimeout, testTokenTimeoutScenarios)
+	})
+
+	updateOAuthConfigInactivityTimeout(t, configClient, nil)
+	test.WaitForKubeAPIServerStartProgressing(t, configClient)
+	test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	// No OAuthClient timeout and no OAuth config timeout.
+	t.Run("unset-inactivity-timeout-client-timeout", func(t *testing.T) {
+		checkTokenAccess(t, userClient, oauthClientClient, &configInactivityTimeout, nil, testTokenTimeoutScenarios)
+	})
+
+}
+
+func checkTokenAccess(t *testing.T,
+	userClient *userclient.UserV1Client,
+	oauthClientClient *oauthclient.OauthV1Client,
+	configInactivityTimeout, oauthClientTimeout *int32,
```

Comment:
why is configInactivityTimeout a pointer?

Reply:
no specific reason. I guess I was just lazy to declare variable of `int32` type